src/main/java/teammates/common/datatransfer/questions/FeedbackMsqQuestionDetails.java

```diff
@@ -756,4 +948,154 @@ private double divideOrReturnZero(double numerator, int denominator) {
         return (denominator == 0) ? 0 : numerator / denominator;
     }
 
+    /**
+     * Calculate the Per Recipient Statistics for MSQ questions.
+     */
+    private static class MsqPerRecipientStatistics {
+        List<String> msqChoices;
+        boolean otherEnabled;
+        McqStatistics msqStats;
+
+        MsqPerRecipientStatistics(FeedbackMsqQuestionDetails msqDetails,
+                McqStatistics msqStats) {
+            this.msqChoices = msqDetails.getMsqChoices();
+            this.otherEnabled = msqDetails.getOtherEnabled();
+            this.msqStats = msqStats;
+        }
+
+        /**
+         * Calculates the number of responses for each response attribute.
+         */
+        private Map<String, Integer> getNumberOfResponsesForEachResponse(FeedbackMsqResponseDetails responseDetails,
+                Map<String, Integer> responseCountPerOption) {
+            List<String> answerStrings = responseDetails.getAnswerStrings();
+            boolean isOtherOptionAnswer = responseDetails.isOtherOptionAnswer();
+            String otherAnswer = "";
+
+            if (isOtherOptionAnswer) {
+                responseCountPerOption.put("Other", responseCountPerOption.getOrDefault("Other", 0) + 1);
+
+                // remove other answer temporarily to calculate stats for other options
+                otherAnswer = answerStrings.get(answerStrings.size() - 1);
```

Comment:
Should otherAnswer be contained in `otherFieldContent`?

Reply:
>Should otherAnswer be contained in otherFieldContent?

I have a little confusion here. Are you saying that we should get the otherAnswer by getting `otherFieldContent` instead of accessing the last element of the answer string? If yes, then we can do that, but we set the `otherFieldContent` using the same mechanism in `FeedbackMsqResponseDetails`

https://github.com/TEAMMATES/teammates/blob/444c0920576e42dbc3d6d6588fc696b500a26d9a/src/main/java/teammates/common/datatransfer/questions/FeedbackMsqResponseDetails.java#L41-L48

Reply:
Better to use `otherFieldContent`. Assuming the last element is the `otherFieldContent` is a bad assumption and maybe we shall remove it in the futhre.